example/src/app/service/hero.service.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { inject, TestBed } from '@angular/core/testing'
+import { HttpClient, HttpClientModule, HttpRequest } from '@angular/common/http'
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing'
+import { heroesUrl, HeroService } from './hero.service'
+
+describe('Service: GoogleBooks', () => {
+  let service: HeroService
+  let backend: HttpTestingController
+
+  const expectedData = {
+    id: 1,
+    name: 'Test hero'
+  }
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientModule,
+        HttpClientTestingModule,
+      ],
+      providers: [
+        HeroService,
+      ],
+    })
+
+    spyOn(console, 'warn')
+    spyOn(console, 'error')
+  })
+
+  beforeEach(inject([ HttpClient, HttpTestingController ],
```

Comment:
Can't we get rid of inline injecting `HttpClient` and `HttpTestingController` here and in `afterEach()` and just move it to the first `beforeEach` call above?

Reply:
ya, it looks cleaner indeed